### PR TITLE
use rustified_enum for ZydisStatusCodes

### PR DESCRIPTION
**Pull request by tathanhdinh**
Hello all,
TL;DR: you may kill me for this PR but let me explain :)

The problem is that when building on Windows, the underlying type of `ZydisStatusCodes` is `i32` (not `u32` as on Linux), this is a [problem](https://github.com/rust-lang-nursery/rust-bindgen/issues/1361) (or feature) of MSVC and not bindgen. So types `ZydisStatus` (i.e. `u32`)  and `ZydisStatusCodes` are not any longer compatible.

Adding some simple type casting `as`, but I find this ugly :) so I've chosen the hard way: use rustified enum for `ZydisStatusCodes`. That leads to a verbose fix, but I find it "more rusty" since `ZydisStatusCodes` is bind really to a enum type in Rust.

Feel free to reject it if you aren't agree :(

NB. Other enums can be rustified too (I've managed to do that for all enums of zydis)

### CODE REVIEW

**Comment by th0rex** [comment]
Hello. Thanks for your PR. I also have noticed this, but I decided to make these changes on the `future` branch, since there are a lot of other things that could be changed on the `master` branch in this repo, but some make use of features in the `future` branch of zydis. Thus I would rather not merge what are essentially the same changes in the master branch here. Also, while you can in fact rustify all the enums on the master branch, it leads to very very ugly usage, namely the `ZydisStatusCodes` having to be written out twice in `ZydisStatusCodes::ZYDIS_STATUS_CODES_SUCCESS` (with `ZydisStatusCodes` just being an example in this case, this applies to all enums). We have talked about this problem a bit, and from what I can tell there are two solutions:

- There already is support for generating most of the "dynamic" enums in another zydis repository, and support for Rust could be added. This would take care of the rather big enums, like the mnemonics. The few remaining enums can be made nicer by hand, as I have done in the future branch (see https://github.com/zyantific/zydis-rs/blob/future/src/bindings.c#L2 for example).
- Add support for "renaming" enums in bindgen, like being able to specify to "drop" the enum names from enum values. This would in my opinion be a much nicer solution and certainly be useful to many other crates as well. However it would not be a trivial change, because there would need to be support for many different naming conventions, i.e. transforming the `ZydisStatusCodes` (name of the enum) to `ZYDIS_STATUS_CODES` (prefix of the enum values) to strip the prefix. I don't know if there already has been such a proposal to the bindgen devs and I wanted to ask what they think about this and then possibly implement it.

Before the next Version of zydis-rs (and zydis) is released one of these two things will be implemented.

Also, I'm sorry that you now have duplicated what we already have, I think the future branch should be mentioned somewhere and I would rather want all new PRs to be against that branch. In fact, I even think that the `future` branch should just be the `master` branch, because in rust there is crates.io and people can get stable versions from there, however the two devs of zydis would rather like that we also follow the zydis way with a stable master branch and a future branch for development, which is also reasonable.

If you want to, you can join the [zydis discord](https://discord.gg/pJaSX3n) or just create some issues here before starting to work on a PR so that we're all on roughly the same page and no ones effort will be wasted.

**Comment by tathanhdinh** [comment]
Wow, thanks a lot @th0rex and sorry for the retard.
I'm instead very happy that the problem will be resolved in the `future` branch.
